File: JavaScript/7js 2/negationValue.js
/*
|--------------------------------------------------------------------------
| Negation of a Value
|--------------------------------------------------------------------------
|
*/

function negationValue(string, value) {
   for (let i = 0; i < string.length; i++) value = !value;
   return value != null ? value : false;
}


function negationValue(string, value) {
   return string.length % 2 == !value;
}


function negationValue(string, value) {
   return string.length & 1 ? !value : !!value;
}


console.log(negationValue("!", false), true);
console.log(negationValue("!", true), false);
console.log(negationValue("!!!", []), false);
console.log(negationValue("", null), false);

File: Python/7py 2/negation_value.py
"""
|--------------------------------------------------------------------------
| Negation of a Value
|--------------------------------------------------------------------------
|
"""

def negation_value(s, val):
   return len(s) % 2 == (not val)


def negation_value(s, val):
   return len(s)&1 ^ bool(val)


def negation_value(str, val):
  return bool(not val if len(str) % 2 else val)


print(negation_value("!", False), True)
print(negation_value("!", True), False)
print(negation_value("!!!", []), True)